tests/unit/s2n_resume_test.c

```diff
@@ -864,6 +1033,23 @@ int main(int argc, char **argv)
             
             EXPECT_SUCCESS(s2n_connection_free(conn));
         }
+
+        /* Fails if keying material expired */
+        {
+            struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+            EXPECT_NOT_NULL(conn);
+            conn->tickets_to_send = UINT16_MAX;
+
+            DEFER_CLEANUP(struct s2n_psk *chosen_psk = s2n_test_psk_new(conn), s2n_psk_free);
+            EXPECT_NOT_NULL(chosen_psk);
+            chosen_psk->type = S2N_PSK_TYPE_RESUMPTION;
+            chosen_psk->keying_material_expiration = 0;
+            conn->psk_params.chosen_psk = chosen_psk;
+
+            EXPECT_FAILURE_WITH_ERRNO(s2n_connection_add_new_tickets_to_send(conn, 1), S2N_ERR_KEYING_MATERIAL_EXPIRED);
```

Comment:
What will happen in the real world if a ticket is added after it expires? Given enough executions this will happen through random chance. (The ticket was valid 50ms ago but now....)

Reply:
We'll generate a ticket that can't actually be used. When the client tries to use it, the check [here](https://github.com/aws/s2n-tls/blob/702428d8893236d97a90a5d0cc7cd21b0c7a5245/tls/s2n_resume.c#L260) will fail, and the ticket will be rejected. That felt like acceptable behavior for me, or at least not any worse than generating a ticket with only 1ns of lifetime left.

If we wanted to try to make both cases less likely, we could add a minimum lifetime, but I'm not sure what we'd want to set that to.